scriptworker/test/test_cot_verify.py

```diff
@@ -1460,6 +1460,61 @@ def fake_commit_msg(*args):
         )
 
 
+@pytest.mark.asyncio
+@pytest.mark.parametrize("push_comment,raises", ((
+    'foo bar baz', False
+), (
+    'try: a b c', False
+), (
+    'blah blah blah blah try: a b c', False
+), (
+    'blah blah blah blah\nlbah blha try: [a] b c\nblah blah', False
+), (
+    'asdfsadfsad', True
+)))
+async def test_get_additional_cron_jsone_context(chain, mocker, push_comment,
+                                                 raises):
+
+    async def fake_pushlog(*args):
+        return {
+            'pushes': {
+                '123': {
+                    'user': 'myuser',
+                    'date': 'mydate',
+                    'changesets': [{
+                        'desc': push_comment
+                    }]
+                }
+            }
+        }
+
+    def fake_commit_msg(*args):
+        return push_comment
+
+    mocker.patch.object(cotverify, 'get_revision', return_value="myrev")
+    mocker.patch.object(cotverify, 'get_pushlog_info', new=fake_pushlog)
+    mocker.patch.object(cotverify, 'get_commit_message', new=fake_commit_msg)
+
+    if raises:
+        with pytest.raises(CoTError):
+            await cotverify._get_additional_cron_jsone_context(
+                chain, chain
```

Comment:
You're using the `chain` fixture, which is a pytest fixture defined [here](https://github.com/mozilla-releng/scriptworker/blob/ebb0a21c445b0ad82d325ba16f46379fdbe883a1/scriptworker/test/test_cot_verify.py#L54). The [task definition](https://github.com/mozilla-releng/scriptworker/blob/ebb0a21c445b0ad82d325ba16f46379fdbe883a1/scriptworker/test/test_cot_verify.py#L57) doesn't contain an `extra` dict. Since `_get_additional_cron_jsone_context` references [`parent_link.task['extra']['cron']`](https://github.com/mozilla-releng/scriptworker/blob/ebb0a21c445b0ad82d325ba16f46379fdbe883a1/scriptworker/cot/verify.py#L1036), you're hitting a `KeyError`.

You may want to use the [`cron_link`](https://github.com/mozilla-releng/scriptworker/blob/ebb0a21c445b0ad82d325ba16f46379fdbe883a1/scriptworker/test/test_cot_verify.py#L192) fixture, which does specify [an empty] `task.extra.cron`.